acceptance-tests/src/main/kotlin/org/ostelco/at/common/StripeEventListener.kt

```diff
@@ -0,0 +1,167 @@
+package org.ostelco.at.common
+
+import com.google.api.gax.core.NoCredentialsProvider
+import com.google.api.gax.grpc.GrpcTransportChannel
+import com.google.api.gax.rpc.FixedTransportChannelProvider
+import com.google.cloud.pubsub.v1.AckReplyConsumer
+import com.google.cloud.pubsub.v1.MessageReceiver
+import com.google.cloud.pubsub.v1.Subscriber
+import com.google.gson.Gson
+import com.google.protobuf.ByteString
+import com.google.pubsub.v1.ProjectSubscriptionName
+import com.stripe.model.Event
+import com.stripe.model.Invoice
+import com.stripe.model.Subscription
+import io.grpc.ManagedChannelBuilder
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeoutOrNull
+
+object StripeEventListener {
+
+    private val logger by getLogger()
+
+    val topic = "stripe-event"
+
+    fun waitForSubscriptionPaymentToSucceed(topic: String = StripeEventListener.topic,
+                                            subscription: String,
+                                            customerId: String,
+                                            timeout: Long = 10000L): Boolean =
+            waitForStripeEvent(
+                    topic = topic,
+                    subscription = subscription,
+                    eventType = "invoice.payment_succeeded",
+                    timeout = timeout,
+                    cmp = {
+                        if (it.type != "invoice.payment_succeeded")
+                            false
+                        else {
+                            val event = it.data.`object` as Invoice
+                            (event.customer == customerId)
+                                    .also {
+                                        if (it)
+                                            logger.info("Received 'invoice.payment_succeeded' event for customer $customerId")
+                                    }
+                        }
+                    })
+
+    fun waitForFailedSubscriptionRenewal(topic: String = StripeEventListener.topic,
```

Comment:
The control flow of this method is very obscure. Please make it simpler to read by breaking it up into smaller methods with descriptive names rather than one big open-coded method with a  very complex internal life.